services/FileService.js

```diff
@@ -0,0 +1,77 @@
+const multer = require("multer");
+const axios = require("axios").default;
+const FormData = require("form-data");
+const { unlinkSync, createReadStream } = require("fs");
+const { extname } = require("path");
+
+const Image = require("../models/Image");
+
+/** Maximum image count */
+const MAX_IMAGE_COUNT = 3;
+/** Destination for storing images */
+const IMAGE_DESTINATION = "./images";
+
+/** Multer Storage Engine, configured to store files locally */
+const storage = multer.diskStorage({
+  destination: IMAGE_DESTINATION,
+  filename: (_, { originalname }, cb) => {
+    cb(null, `${originalname}${extname(originalname)}`);
+  },
+});
+
+/**
+ * Express middleware for storing images on local storage. Saves at max
+ * `maxImages` files
+ */
+const saveImages = multer({
+  storage: storage,
+}).array("images", MAX_IMAGE_COUNT);
+
+/**
+ * Uploads given images to remote server
+ * @param {Express.Multer.File[]} images images to be uploaded
+ * @param {String} userId ID of the user storing the files
+ */
+const uploadImages = async (images, userId) => {
+  const form = new FormData();
+  form.append("userId", userId);
+  images.forEach(({ path }) => form.append("images", createReadStream(path)));
+
+  try {
+    const response = await axios.post(
+      `${process.env.FILE_SERVER_URI}/image`,
+      form,
+      {
+        headers: form.getHeaders(),
+      }
+    );
+
+    const { data, status } = response;
+    if (status === 200) {
+      return data;
+    }
+    return null;
+  } catch (error) {
+    throw error;
```

Comment:
throwing error, if not handled later might crash the system. Are you catching it at a later stage?